woocommerce/woocommerce-ios
# [Woo POS][Cash & Receipts] Adapt send receipt modal to full screen

Closes: #14480

## Description
This PR updates the "send receipt" modal view to a full-screen view, and simplifies its design:

https://github.com/user-attachments/assets/e9981cef-7c07-4fe8-a5ad-6d936f2b3a50

## Testing
Better tested with a physical device, otherwise we can switch the built-in keyboard within the simulator, under I/O > Keyboard > Toggle keyboard

- Set feature flag `.sendReceiptsForPointOfSale` to true
- Log into Povilas testing site https://lovely-aware-sturgeon.jurassic.ninja/ details: PdfdoF-5Qj-p2
- Proceed with a POS transaction
- Tap on the "Receipt" button, observe we switch the view state to the new "send receipt" view
- Observe that the focus is set to the textfield
- Enter an email address, tap "send"
- Observe that the text is changed to a loading indicator while the receipt is being requested

<img width="704" alt="Screenshot 2024-12-10 at 12 52 13" src="https://github.com/user-attachments/assets/9c1150f9-bf52-4afe-a9dd-ff9eacd854a2">


Once the email is sent, the view should be switched back to "TotalsView" automatically.

At the moment there is no email validation, or error handling, so the flow will end successfully despite entering an invalid email address or if there's a receipt error. This will be tackled separately.

---

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:
